### PR TITLE
docs(forms): update form builder API reference

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/forms/src/form_builder.ts [hunk 1]
```diff
@@ -13,44 +13,48 @@ import {AbstractControl, FormArray, FormControl, FormGroup} from './model';
...
+
   /**
-   * Construct a new `FormControl` with the given `formState`,`validator`, and
-   * `asyncValidator`.
+   * @description
+   * Construct a new `FormControl` instance.
+   *
+   * @param formState Initializes the control with an initial state value,
+   * or with an object that defines the initial value, status, and options
+   * for handling updates and validation.
```
Comment (kara) [inline packages/forms/src/form_builder.ts]: The formState doesn't actually support status and options. You can either provide an initial value, or a boxed value that is wrapped with the disabled value (e.g. `{value: 'some value', disabled: true}`).

Note: It looks like this is actually wrong in the `FormControl` documentation as well. Can you fix there too?

**Review by kara — CHANGES_REQUESTED** [review]

File: packages/forms/src/form_builder.ts (hunk 1 above)
Comment (brandonroberts) [reply packages/forms/src/form_builder.ts]: I see. Yes, I'll update that one also

Comment (brandonroberts) [reply packages/forms/src/form_builder.ts]: Is this more accurate?


```
Initializes the control with an initial value or a boxed value that defines the initial and disabled values.
```

**Comment by brandonroberts** [comment]
@kara review comments have been addressed

File: packages/forms/src/form_builder.ts [hunk 2]
```diff
@@ -13,44 +13,47 @@ import {AbstractControl, FormArray, FormControl, FormGroup} from './model';
...
   }
+
   /**
-   * Construct a new `FormControl` with the given `formState`,`validator`, and
-   * `asyncValidator`.
+   * @description
+   * Construct a new `FormControl` instance.
+   *
+   * @param formState Initializes the control with an initial value,
+   * or a boxed value that defines the initial value and disabled state.
```
Comment (kara) [inline packages/forms/src/form_builder.ts]: Nit: Can we provide a short example of what the boxed value looks like? 

> e.g. {value: 'my val', disabled: true}

Comment (brandonroberts) [reply packages/forms/src/form_builder.ts]: Done

File: packages/examples/forms/ts/formBuilder/form_builder_example.ts [hunk 3]
```diff
@@ -40,3 +41,19 @@ export class FormBuilderComp {
...
+@Component({
+  selector: 'app-disabled-form-control',
+  template: `
+    <input [formControl]="control" placeholder="First">
+  `
+})
+export class DisabledFormControlComponent {
+  control: FormControl;
+
+  constructor(@Inject(FormBuilder) fb: FormBuilder) {
```
Comment (kara) [inline packages/examples/forms/ts/formBuilder/form_builder_example.ts]: Is this `@Inject` necessary here?

File: packages/forms/src/form_builder.ts [hunk 4]
```diff
@@ -13,44 +13,57 @@ import {AbstractControl, FormArray, FormControl, FormGroup} from './model';
...
-   * `formState` can either be a standalone value for the form control or an object
-   * that contains both a value and a disabled status.
+   * @param validator A synchronous validator function, or an array of synchronous validator
+   * functions.
+   *
+   * @param asyncValidator A single async validator or array of async validator functions
+   *
+   * @usageNotes
+   *
+   * ### Initial a control as disabled
```
Comment (kara) [inline packages/forms/src/form_builder.ts]: Initial -> Initialize

File: packages/forms/src/form_builder.ts [hunk 5]
```diff
@@ -60,8 +73,16 @@ export class FormBuilder {
   }
 
   /**
-   * Construct a `FormArray` from the given `controlsConfig` array of
-   * configuration, with the given optional `validator` and `asyncValidator`.
+   * @description
+   * Construct a new `FormArray` instance.
+   *
+   * @param controlsConfig A array of child controls. The key for each child control is its index in
```
Comment (kara) [inline packages/forms/src/form_builder.ts]: A array -> An array

**Review by kara — COMMENTED** [review]
Few more typos

File: packages/examples/forms/ts/formBuilder/form_builder_example.ts (hunk 3 above)
Comment (brandonroberts) [reply packages/examples/forms/ts/formBuilder/form_builder_example.ts]: No, its not. Removed

File: packages/forms/src/form_builder.ts [hunk 6]
```diff
@@ -60,8 +73,17 @@ export class FormBuilder {
...
 
   /**
-   * Construct a `FormArray` from the given `controlsConfig` array of
-   * configuration, with the given optional `validator` and `asyncValidator`.
+   * @description
+   * Construct a new `FormArray` instance.
+   *
+   * @param controlsConfig An array of child controls. The key for each child control is its index
+   * in
+   * the array.
```
Comment (kara) [inline packages/forms/src/form_builder.ts]: Nit: fix formatting?

**Review by kara — APPROVED** [review]
LGTM

**Pull request closed without merging**